src/types/viewport.cpp

```diff
@@ -432,7 +433,7 @@ bool Viewport::WalkInBounds(til::point& pos, const WalkDir dir, bool allowEndExc
 bool Viewport::WalkInBoundsCircular(til::point& pos, const WalkDir dir, bool allowEndExclusive) const noexcept
 {
     // Assert that the position given fits inside this viewport.
-    FAIL_FAST_IF(!IsInBounds(pos, allowEndExclusive));
+    assert(IsInBounds(pos, allowEndExclusive));

```

Comment:
What happens down the line when UIA _tries to use this new coordinate_ that is provably out of bounds? What happens when we Walk it in bounds circular?

Reply:
Hmmmmmmm.

I guess the problem with `MyClamp` is that _if we knew when to use it,_` we wouldn't have had this problem in the first place. Right?

Reply:
I personally would say that we should use fail-fasts or exceptions for security or safety relevant checks (e.g. boundary checks for array access). For anything else I'd use debug assertions + clamping/truncation/etc. in release on a "lazy" best-effort basis.

The suggested code above would be overkill already for that purpose IMO. It's not just stopping incorrect behavior, but also tries to correct the position in a smart manner. But how do we know that this improves the behavior? It shouldn't pass incorrect positions in the first place after all!

As such I'd just use `std::clamp` 4 times for numeric parameters like that.